### PR TITLE
Marcuse/6696

### PR DESCRIPTION
**Pull request by ptnapoleon**
Opening PR here to consolidate review

Depends on https://github.com/pcmanus/ccm/pull/433

### CODE REVIEW

File: compaction_test.py [hunk 1]
```diff
@@ -121,12 +121,13 @@ def sstable_deletion_test(self):
             session.execute('delete from cf where key = ' + str(x))
 
         block_on_compaction_log(node1, ks='ks', table='cf')
-
+        time.sleep(1)
```
Comment (ptnapoleon) [inline compaction_test.py]: Why add the sleep?

**Comment by ptnapoleon** [comment]
Other than the travis complaints, and the new sleeps, I'm +1. I'm running on CI again now. Have you run it against your branch?

Won't we have issues that we're defaulting to 3 data dirs, but 6696 is only on 3.2+?

**Comment by ptnapoleon** [comment]
I'm seeing
```
cassandra/bin/nodetool -h localhost -p 7100 rebalancedisks' failed; exit status: 1; stdout: nodetool: Found unexpected parameters: [rebalancedisks]
```
in blacklisted_directory_test (disk_balance_test.TestDiskBalance) against your branch

**Comment by ptnapoleon** [comment]
These changes break `dtcs_deletion_test(compaction_test.TestCompaction_with_DateTieredCompactionStrategy)` on 2.1

File: disk_balance_test.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+class TestDiskBalance(Tester):
+
+    def disk_balance_stress_test(self):
+        cluster = self.cluster
+        cluster.set_datadir_count(3)
+        cluster.set_configuration_options(values={'allocate_tokens_for_keyspace': 'keyspace1'})
+        cluster.populate(4).start(wait_for_binary_proto=True)
+        [node1, node2, node3, node4] = cluster.nodelist()
+
+        node1.stress(['write', 'n=10000', '-rate', 'threads=100', '-schema', 'replication(factor=2)'])
```
Comment (ptnapoleon) [inline disk_balance_test.py]: `n=10k` please

File: disk_balance_test.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+
+class TestDiskBalance(Tester):
+
+    def disk_balance_stress_test(self):
+        cluster = self.cluster
+        cluster.set_datadir_count(3)
+        cluster.set_configuration_options(values={'allocate_tokens_for_keyspace': 'keyspace1'})
+        cluster.populate(4).start(wait_for_binary_proto=True)
+        [node1, node2, node3, node4] = cluster.nodelist()
```
Comment (ptnapoleon) [inline disk_balance_test.py]: No need for the brackets on the left half of this expression

File: disk_balance_test.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
+import os
+import os.path
+from dtest import Tester, debug, DISABLE_VNODES
+from tools import since, new_node, create_c1c2_table, insert_c1c2, query_c1c2, no_vnodes
+from assertions import assert_almost_equal
+from jmxutils import JolokiaAgent, make_mbean, remove_perf_disable_shared_mem
+from cassandra import ConsistencyLevel
+
+
+class TestDiskBalance(Tester):
```
Comment (ptnapoleon) [inline disk_balance_test.py]: Shouldn't this be restricted to only run on 3.2+?

File: disk_balance_test.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        # make sure the data directories are balanced:
+        for node in cluster.nodelist():
+            self.assert_balanced(node)
+
+    def disk_balance_bootstrap_test(self):
+        cluster = self.cluster
+        cluster.set_datadir_count(3)
+        cluster.set_configuration_options(values={'allocate_tokens_for_keyspace': 'keyspace1'})
+        cluster.populate(4).start(wait_for_binary_proto=True)
+        [node1, node2, node3, node4] = cluster.nodelist()
```
Comment (ptnapoleon) [inline disk_balance_test.py]: Remove brackets.

File: disk_balance_test.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+        node1.stress(['write', 'n=10000', '-rate', 'threads=100', '-schema', 'replication(factor=2)'])
```
Comment (ptnapoleon) [inline disk_balance_test.py]: `n=10k`

File: disk_balance_test.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+        for node in cluster.nodelist():
+            self.assert_balanced(node)
+
+    def disk_balance_decommission_test(self):
+        cluster = self.cluster
+        cluster.set_datadir_count(3)
+        cluster.set_configuration_options(values={'allocate_tokens_for_keyspace': 'keyspace1'})
+        cluster.populate(4).start(wait_for_binary_proto=True)
+        [node1, node2, node3, node4] = cluster.nodelist()
```
Comment (ptnapoleon) [inline disk_balance_test.py]: brackets

File: disk_balance_test.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        node1.stress(['write', 'n=1', '-rate', 'threads=100', '-schema', 'replication(factor=2)'])
+        for node in cluster.nodelist():
+            node.nodetool('disableautocompaction')
+
+        node1.stress(['write', 'n=10000', '-rate', 'threads=100', '-schema', 'replication(factor=2)'])
```
Comment (ptnapoleon) [inline disk_balance_test.py]: 10k

File: disk_balance_test.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        cluster = self.cluster
+        cluster.set_datadir_count(3)
+        cluster.set_configuration_options(values={'allocate_tokens_for_keyspace': 'keyspace1'})
+        cluster.populate(3).start(wait_for_binary_proto=True)
+        [node1, node2, node3] = cluster.nodelist()
+        node1.stress(['write', 'n=1', '-rate', 'threads=100', '-schema', 'replication(factor=1)'])
+        cluster.flush()
+        session = self.patient_cql_connection(node1)
+        session.execute("ALTER KEYSPACE keyspace1 WITH replication = {'class':'SimpleStrategy', 'replication_factor':2}")
+        node1.stress(['write', 'n=100000', '-rate', 'threads=100'])
```
Comment (ptnapoleon) [inline disk_balance_test.py]: `n=100k`

File: disk_balance_test.py [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+        for k in xrange(0, 10000):
+            query_c1c2(session, k)
+
+    def alter_replication_factor_test(self):
+        cluster = self.cluster
+        cluster.set_datadir_count(3)
+        cluster.set_configuration_options(values={'allocate_tokens_for_keyspace': 'keyspace1'})
+        cluster.populate(3).start(wait_for_binary_proto=True)
+        [node1, node2, node3] = cluster.nodelist()
```
Comment (ptnapoleon) [inline disk_balance_test.py]: Brackets, again, thanks

File: disk_balance_test.py (hunk 4 above)
Comment (ptnapoleon) [inline disk_balance_test.py]: Could you add a docstring that looks like
```
"""
@jira_ticket CASSANDRA-6696
"""
```

**Comment by ptnapoleon** [comment]
I think this also may be breaking `secondary_indexes_test.TestSecondaryIndexes.test_manual_rebuild_index` on 3.0

**Comment by krummas** [comment]
I think I fixed the issues now, and triggered 2.1, 2.2 and 3.0 cassci builds here:
https://cassci.datastax.com/view/Dev/view/krummas/job/krummas-marcuse-2.1-dtest/
https://cassci.datastax.com/view/Dev/view/krummas/job/krummas-marcuse-2.2-dtest/
https://cassci.datastax.com/view/Dev/view/krummas/job/krummas-marcuse-3.0-dtest/

we now need the sleep as it takes a bit longer to remove the sstables from several data directories

**Comment by ptnapoleon** [comment]
You can check here to see the failures that aren't on the main branches:
http://cassci.datastax.com/userContent/cstar_report/index.html?jobs=krummas-marcuse-2.1-dtest,krummas-marcuse-2.2-dtest,krummas-marcuse-3.0-dtest,cassandra-2.1_dtest

Looks like a few related to these changes.

**Comment by ptnapoleon** [comment]
I see no new failures after having run this without vnodes:
http://cassci.datastax.com/userContent/cstar_report/index.html?jobs=krummas-marcuse-2.1-dtest,krummas-marcuse-2.2-dtest,krummas-marcuse-3.0-dtest,cassandra-2.1_dtest,krummas-marcuse-2.1-novnode-dtest,cassandra-2.1_novnode_dtest

**Pull request merged**